### PR TITLE
ability to make a file executable

### PR DESCRIPTION
**Pull request by jessitron**
i need this for the editor I'm working on now.
Opted to implement it async for consistency with the other methods I'm using. 
the isExecutable exists to make the other testable

### CODE REVIEW

File: src/project/local/NodeFsLocalFile.ts [hunk 1]
```diff
@@ -47,6 +48,10 @@ export class NodeFsLocalFile extends AbstractFile implements LocalFile {
         }
         return Promise.resolve(this);
     }
+
+    public isExecutable(): Promise<boolean> {
+        return fs.access(this.realPath, fs.constants.X_OK).then(_ => true).catch(_ => false);
```
Comment (cdupuis) [inline src/project/local/NodeFsLocalFile.ts]: why those underscores? if you don't need use the variables use `()` instead. `_` for throw-away parameters is not idiomatic for JS.

File: src/project/local/NodeFsLocalProject.ts [hunk 2]
```diff
@@ -129,6 +130,15 @@ export class NodeFsLocalProject extends AbstractProject implements LocalProject
         return fs.unlink(this.toRealPath(path)).then(_ => this);
     }
 
+    public makeExecutable(path: string): Promise<this> {
+        return fs.stat(this.toRealPath(path))
+            .then(stats => {
+                logger.debug("Starting mode: " + stats.mode);
+                return fs.chmod(this.toRealPath(path), stats.mode && fs.constants.S_IXUSR);
+            } )
+            .then(_ => this);
```
Comment (cdupuis) [inline src/project/local/NodeFsLocalProject.ts]: see above.

**Comment by johnsonr** [comment]
LGTM. Agree with @cdupuis comment on `_`

**Comment by cdupuis** [comment]
Pull request auto merge enabled. [atomist:enable-auto-merge]

**Comment by slimslenderslacks** [comment]
Pull request auto merged by Atomist.

* 1 approved review by @cdupuis
* 4 successful checks

**Pull request merged**